agent/src/com/thoughtworks/go/agent/AgentWebSocketClientController.java

```diff
@@ -166,7 +166,8 @@ private void runBuild(BuildSettings buildSettings) {
             buildConsole = new ConsoleOutputTransmitter(
                 new RemoteConsoleAppender(
                     urlService.prefixPartialUrl(buildSettings.getConsoleUrl()),
-                    httpService)
+                    httpService,
+                    getSystemEnvironment().consoleLogCharsetAsCharset())
```

Comment:
This would be set at the server side not agent. The charset that was passed along from the server to the agent needs to be passed along to the Console appender